packages/apputils/src/printing.ts

```diff
@@ -0,0 +1,116 @@
+/**
+ * Any object is printable if it implements the `IPrintable` interface.
+ *
+ * If you have a custom widget and would like to make it printable, then you
+ * should fulfil that interface on that widget by adding a `[printSymbol]` method.
+ *
+ * One way of printing is to use the `printWidget` function, which creates a hidden iframe
+ * and copies the DOM nodes from your widget to that iframe and printing just that iframe.
+ *
+ * To use this method in your widget, set the `[printSymbol]` method to `printWidget`.
+ * You can also provide custom args to that function, including CSS text to apply
+ * when printing.
+ *
+ * See https://github.com/joseluisq/printd#print
+ *
+ * Another way to print is to use the `printURL` function, which takes a URL and prints that page.
+ */
+
+import { Printd, PrintdCallback } from 'printd';
+import { Widget } from '@phosphor/widgets';
+
+/**
+ * Symbol to use for a method that prints out out the object.
+ */
+export const printSymbol = Symbol();
+
+/**
+ * Widgets who provide a custom way of printing themselves
+ * should implement this interface.
+ */
+export interface IPrintable {
+  [printSymbol]: () => void;
+}
+
+/**
+ * Returns whether an object implements a print method.
+ */
+export function isPrintable(a: any): a is IPrintable {
```

Comment:
Should we type `a: object` to get a compile-time check as well as a runtime check?

Reply:
Good call! I made it [`unknown`](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-0.html#new-unknown-top-type) which is even safer to use than `object` (i.e. it could be a primitive type).